### PR TITLE
Fix #1409 installerOptions.failure() options can be undefined

### LINKED ISSUE #1409: installerOptions.failure() options can be undefined
### Description

As the title says and contrary to the type definitions, the `failure()` callback within `installerOptions` can receive `undefined`. Specifically I got that when aborting the OAuth flow.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] example code related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/bolt/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Reproducible in:

package version: 3.8.1

node version: 16.8

OS version(s): MacOS

#### Steps to reproduce:

1. Set-up `installerOptions.failure(error, options) { console.log({ options }); }`
2. Use the associated `installer.generateInstallUrl()` to go to the oauth flow
3. Click "Cancel"

#### Expected result:

Receive an `options` object with metadata etc.

#### Actual result:

Received `undefined`

### PR DESCRIPTION
**Pull request by seratch**
###  Summary

This pull request resolves #1409 

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/oauth/src/index.spec.js [hunk 1]
```diff
@@ -566,7 +569,6 @@ describe('OAuth', async () => {
         },
         failure: async (error, installOptions, req, res) => {
           assert.fail(error.message);
-          res.send('failure');
```
Comment (seratch) [inline packages/oauth/src/index.spec.js:569]: This is just unnecessary

File: packages/oauth/src/index.spec.js [hunk 2]
```diff
@@ -588,7 +590,6 @@ describe('OAuth', async () => {
...
```
Comment (seratch) [inline packages/oauth/src/index.spec.js:591]: same as above

File: packages/oauth/src/index.ts [hunk 3]
```diff
@@ -515,13 +515,20 @@ export class InstallProvider {
...
+        // To make the `installOptions` type compatible with `CallbackOptions#failure` signature
+        const emptyInstallOptions: InstallURLOptions = { scopes: [] };
+        // eslint-disable-next-line no-param-reassign
+        installOptions = emptyInstallOptions;
+      }
+
       // Call the failure callback
       if (options !== undefined && options.failure !== undefined) {
         this.logger.debug('calling passed in options.failure');
-        options.failure(error, installOptions!, req, res);
```
Comment (seratch) [inline packages/oauth/src/index.ts:521]: Removing `!` here now compiles!

File: packages/oauth/src/index.spec.js [hunk 4]
```diff
@@ -447,6 +447,7 @@ describe('OAuth', async () => {
           assert.fail('should have failed');
         },
         failure: async (error, installOptions, req, res) => {
+          assert.isDefined(installOptions);
```
Comment (seratch) [inline packages/oauth/src/index.spec.js:450]: This fails before the changes in this PR

**Review by seratch — COMMENTED** [review]
comments for reviewers

File: packages/oauth/src/index.spec.js (hunk 4 above)
Comment (filmaj) [reply packages/oauth/src/index.spec.js:450]: May I also suggest updating the test title (here and for the other tests) to reflect this new requirement? Perhaps changing "should call the failure callback..." to "should call the failure callback with installation options..."?

**Review by filmaj — APPROVED** [review]
LGTM! Left a tiny comment.

File: packages/oauth/src/index.spec.js (hunk 4 above)
Comment (seratch) [reply packages/oauth/src/index.spec.js:450]: Great suggestion -- will update the test label before merging this

**Pull request merged**